### ISSUE TITLE
Load config as a JS file only. Allow dynamic logic/options

### ISSUE THREAD

**Issue by ballercat** [ballercat/jambox#34]
Fully switch over to using the config as a loaded js module. Add some dynamic option support, the two main features are listed below.

## `cache.hashFunction(request, defaultHashFunction) -> Promise<hash>`

Let the user define a custom hash function. Very useful for certain type of requests. The resulting `hash` is used throughout for identifying the cached record (like filenames, ids). A user could for example always return the same hash value if desired for certain type of requests which maintain dynamic parameters but don't influence the response. A custom `hash` value can also make artifacts easily identifiable.

The hash function will always be preferred over `defaultHashFunction`.

## `stub: (request) -> Promise<response>`

Let the user define custom stubs via a function. The glob matching might be too difficult.

**Comment by hamlim** [comment]
Do you think the `stub` change should be per-pattern also, or only either a function or a record?

e.g.

```js
// jambox config
module.exports = {
  stub: {
    // "dynamic" stub:
    '**/log': (request) => {
      return Promise.resolve({ ok: true });
    },
    // "static" stub:
    '**/*.png': {
      status: 200,
      preferNetwork: true,
      file: path.join(__dirname, '.jambox', 'placeholder.png'),
    },
  }
}
```

_Or:_

```js
// jambox config
module.exports = {
  stub: request => {
    if (matches(request.url, '**/log')) {
      return Promise.resolve({ ok: true });
    } else if (matches(request.url, '**/.png')) {
      return {
        status: 200,
        preferNetwork: true,
        file: path.join(__dirname, '.jambox', 'placeholder.png'),
      };
    }
    return Promise.reject({notStubbed: true});
  }
}
```

**Comment by ballercat** [comment]
@hamlim 

Strong preference for the latter. Though to be fair, both versions could be allowed. One function for all stubs supports every possible scenario so it would at least make sense as the first take.

**Comment by ballercat** [comment]
Yeah sure that can be done. Is it worth it? I don't know, it's not _difficult_ to provide it, but its also might not end-up being very useful
